ArduPlane/system.cpp

```diff
@@ -240,6 +240,19 @@ bool Plane::set_mode(Mode &new_mode, const ModeReason reason)
     }
 #endif  // !QAUTOTUNE_ENABLED
 
+    // check if we should refuse auto mode due to a missing takeoff in
+    // guided_wait_takeoff state
+    if (&new_mode == &plane.mode_auto &&
+        control_mode == &plane.mode_guided &&
+        quadplane.guided_wait_takeoff) {
+        uint16_t cmd_id;
+        if (!mission.get_next_nav_cmd_id(cmd_id) ||
+            !quadplane.is_vtol_takeoff(cmd_id)) {
+            gcs().send_text(MAV_SEVERITY_ERROR,"Takeoff waypoint required");
+            return false;
+        }
+    }
+    
```

Comment:
can't this go in the mode_auto `_enter` function? You should be able to return false from it.

Reply:
yes, it can, i've moved it to the enter function.